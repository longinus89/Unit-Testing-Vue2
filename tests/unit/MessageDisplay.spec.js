import MessageDisplay from '@/components/MessageDisplay'
import { mount } from '@vue/test-utils'
import { getMessage } from '@/services/axios'
import flushPromises from 'flush-promises'

jest.mock('@/services/axios')


describe('MessageDisplay', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })
  it('Calls getMessage once and displays message', async () => {
    const mockMessage = 'Hello from the db'
    getMessage.mockResolvedValueOnce(mockMessage)
    const wrapper = mount(MessageDisplay)

    await flushPromises()
    expect(getMessage).toHaveBeenCalledTimes(1) // check that call happened once
    // check that component displays message
    expect(wrapper.find('p[data-testid="message"]').element.textContent).toMatchSnapshot();
  })

  it('Calls getMessage once, fails and and displays message', async () => {
    const mockError = 'Oops! Something went wrong.'
    getMessage.mockRejectedValueOnce(mockError)
    const wrapper = mount(MessageDisplay)

    await flushPromises()
    expect(getMessage).toHaveBeenCalledTimes(1) // check that call happened once
    // check that component displays message
    expect(wrapper.find('p[data-testid="message-error"]').element.textContent).toMatchSnapshot();
  })
})