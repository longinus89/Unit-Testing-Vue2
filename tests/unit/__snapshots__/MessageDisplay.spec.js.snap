// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageDisplay Calls getMessage once and displays message 1`] = `"Hello from the db"`;

exports[`MessageDisplay Calls getMessage once, fails and and displays message 1`] = `"Oops! Something went wrong."`;
